format.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+#
+# format.sh
+#
+# run clang-format on each .c & .h file
+
+if [ -z "${CLANG_FORMAT}" ]; then
+    CLANG_FORMAT=clang-format-3.9
+fi
+
+a=`git ls-files | grep "\.h$\|\.c$"`
```

Comment:
would `git ls-files '*.c' '*.h'` work just as well?

Reply:
no but `git ls-files *.c *.h` seams to will change it later. Plan to make the script report that code was changed so can be used as part of a travis script.

Reply:
Maybe the single quote mark needs to be escaped.  My concern with having a raw *.c there is that the shell might expand that to be whatever .c file you have in the current directory.  (We don't have any in the root directory, but what if one is there?